Sources/BraintreeCoreSwift/BTHTTP.swift

```diff
@@ -0,0 +1,542 @@
+// TODO: - Mark interval vs private properties accordingly
+    public typealias RequestCompletion = (BTJSON?, HTTPURLResponse?, Error?) -> Void
+
+    // MARK: - Public Properties
+    /// An array of pinned certificates, each an NSData instance consisting of DER encoded x509 certificates
+    public let pinnedCertificates: [NSData]
+
+    // TODO: Make internal with Swift test?
+    /// Session exposed for testing
+    public lazy var session: URLSession = {
+        let configuration: URLSessionConfiguration = URLSessionConfiguration.ephemeral
+        configuration.httpAdditionalHeaders = defaultHeaders()
+        
+        let delegateQueue: OperationQueue = OperationQueue()
+        delegateQueue.name = "com.braintreepayments.BTHTTP"
+        delegateQueue.maxConcurrentOperationCount = OperationQueue.defaultMaxConcurrentOperationCount
+        
+        return URLSession(configuration: configuration, delegate: self, delegateQueue: delegateQueue)
+    }()
+
+    // TODO: Make internal with Swift test?
+    /// DispatchQueue on which asynchronous code will be executed. Defaults to `DispatchQueue.main`.
+    public var dispatchQueue: DispatchQueue = DispatchQueue.main
+
+    // MARK: - Internal Properties
+    let cacheDateValidator: BTCacheDateValidator
+    let baseURL: URL
+
+// TODO: - once BTAPIHTTP + BTGraphQLHTTP are converted to Swift, handle error cases for empty or nil credentials before API request is made
+    var authorizationFingerprint: String? = ""
+    var tokenizationKey: String? = ""
+    
+    // MARK: - Initializers
+
+    /// Initialize `BTHTTP` with the URL for the Braintree API
+    /// - Parameter url: The base URL for the Braintree Client API
+    @objc(initWithBaseURL:)
+    public init(url: URL) {
+        self.baseURL = url
+        self.cacheDateValidator = BTCacheDateValidator()
+        self.pinnedCertificates = BTAPIPinnedCertificates.trustedCertificates()
+    }
+
+    /// Initialize `BTHTTP` with the URL from Braintree API and the authorization fingerprint from a client token
+    /// - Parameters:
+    ///   - url: The base URL for the Braintree Client API
+    ///   - authorizationFingerprint: The authorization fingerprint HMAC from a client token
+    @objc(initWithBaseURL:authorizationFingerprint:)
+    public convenience init(url: URL, authorizationFingerprint: String) {
+        self.init(url: url)
+        self.authorizationFingerprint = authorizationFingerprint
+    }
+
+    /// Initialize `BTHTTP` with the URL from Braintree API and the authorization fingerprint from a tokenizationKey
+    /// - Parameters:
+    ///   - url: The base URL for the Braintree Client API
+    ///   - tokenizationKey: The authorization fingerprint HMAC from a client token
+    @objc(initWithBaseURL:tokenizationKey:)
+    public convenience init(url: URL, tokenizationKey: String) {
+        self.init(url: url)
+        self.tokenizationKey = tokenizationKey
+    }
```

Comment:
For this and the previous `convenience init` can we do something like this?

```suggestion
    @objc(initWithBaseURL:tokenizationKey:error)
    public convenience init(url: URL, tokenizationKey: String) throws {
        if tokenizationKey.isEmpty {
            throw BTHTTPError.invalidTokenizationKey
        }
        self.init(url: url)
        self.tokenizationKey = tokenizationKey
    }
```

Reply:
maybe we could use some kind of enum to determine whether we're using auth fingerprint or tokenization key?